libsolidity/ast/Types.cpp

```diff
@@ -1570,9 +1570,9 @@ BoolResult ArrayType::isExplicitlyConvertibleTo(Type const& _convertTo) const
 {
 	if (isImplicitlyConvertibleTo(_convertTo))
 		return true;
-	// allow conversion bytes <-> string
+	// allow conversion bytes <-> string and bytes -> bytesNN
 	if (_convertTo.category() != category())
-		return false;
+		return isByteArray() && !isString() && _convertTo.category() == Type::Category::FixedBytes;
```

Comment:
Is `isByteArray()` true for `bytes1[]`?

Reply:
no, it is regular array type of `bytes1`